docs/design/chart_format.md

```diff
@@ -0,0 +1,664 @@
+
+For the purpose of providing versioning ranges, and also for backward compatibility, version requirements may be treated as as suffix condition, instead of part of the path:
+
+```
+helm:example.com/charts/nginx#1.2.3   // Exact version
+helm:example.com/charts/nginx#~2.1    // Version range
+helm:example.com/charts/nginx         // Latest release
+```
+
+The first of the above two short names is equivalent to helm:example.com/charts/nginx-1.2.3.tgz. The second example uses a semantic version filter ~1.2, which means “>=1.2.0, <1.3.0”, or “any patch version of 1.2”. Other filters include the “^” operator (^1 is “any minor version in the 1.x line), and “>”, “>=”, “=”, “<”, and “<=” operators. These are standard SemVer filter operators.
+
+
+Any short form handler should be able to resolve the default short form as specified above.
+
+
+### Local References
+During chart development, and in other special circumstances, it may be desirable to work with an unversioned, unpackaged local copy of a chart. For the sake of consistency across products, an explicit naming formula is to be followed.
+
+
+In cases where a local path is used in lieu of a full or short name, the path string _must_ begin with either a dot (.) or a slash (/) or the file schema(`file`:).
+
+
+Legal examples of this include:
+```
+  ./foo
+  ../foo
+  /foo
+  //foo
+  /.foo
+  file:///example/foo
+```
+
+Unprefixed relative paths are not valid. For example, `foo/` is not allowed as a local path, as it conflicts with a legal short name, and is thus ambiguous.
+
+
+
+## Appendix A: User Stories for Charts
+Personas:
+
+
+- Operator: Responsible for running an application in production.
+- Service Dev: Responsible for developing new definitions
+- App Dev: Developer who creates applications that make use of existing definitions, but does not create definitions.
+
+
+Stories:
+
+
+- As an operator, I want a build that is 100% reproducible (exact versions)
+- As an app dev, I want to be able to search for definitions...
+        * by keyword, where one app may have multiple keywords (e.g. Redis has storage, message queue)
+        * by name, where name may be "fuzzy"
+        * by author
+        * by last updated date
+- As a service dev, I want a well-defined set of practices to follow
+- As a service dev, I want to be able to work with a team on the same definition
+- As a service dev, I want to be able to indicate when a particular definition is stable, and how stable it is
+- As a service dev, I want to indicate the role I played in building a definition
+- As a service dev, I want to be able to use all of the low-level Kubernetes kinds
+- As an operator, I want to be able to determine how stable a package is
+- As an operator, I want to be able to determine what version of Kubernetes I need to run a definition
+- As an operator, I want to determine whether a definition requires extension kinds (e.g. DaemonSet or something custom), and determine this _before_ I try to install
```

Comment:
We should check the required extensions before deploying, not require the user to check it. This functionality is not called out earlier in the document, but should be. I have also opened an issue for it to make sure it gets included in the release (#252).